registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -242,35 +278,34 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeAngleDerivatives
-    angular_hessian_.row(3).noalias() = Eigen::Vector4d((sx * cy * cz), (-sx * cy * sz), (sx * sy), 0.0f); // b3
+    angular_hessian_.row(2).noalias() = Eigen::Vector4d((cx * cy * cz), (-cx * cy * sz), (cx * sy), 0.0); // b2
+    angular_hessian_.row(3).noalias() = Eigen::Vector4d((sx * cy * cz), (-sx * cy * sz), (sx * sy), 0.0); // b3
 
-    angular_hessian_.row(4).noalias() = Eigen::Vector4d((-sx * cz - cx * sy * sz), (sx * sz - cx * sy * cz), 0, 0.0f); // c2
-    angular_hessian_.row(5).noalias() = Eigen::Vector4d((cx * cz - sx * sy * sz), (-sx * sy * cz - cx * sz), 0, 0.0f); // c3
+    angular_hessian_.row(4).noalias() = Eigen::Vector4d((-sx * cz - cx * sy * sz), (sx * sz - cx * sy * cz), 0.0, 0.0); // c2
+    angular_hessian_.row(5).noalias() = Eigen::Vector4d((cx * cz - sx * sy * sz), (-sx * sy * cz - cx * sz), 0.0, 0.0); // c3
 
-    angular_hessian_.row(6).noalias() = Eigen::Vector4d((-cy * cz), (cy * sz), (sy), 0.0f); // d1
-    angular_hessian_.row(7).noalias() = Eigen::Vector4d((-sx * sy * cz), (sx * sy * sz), (sx * cy), 0.0f); // d2
-    angular_hessian_.row(8).noalias() = Eigen::Vector4d((cx * sy * cz), (-cx * sy * sz), (-cx * cy), 0.0f); // d3
+    angular_hessian_.row(6).noalias() = Eigen::Vector4d((-cy * cz), (cy * sz), (sy), 0.0); // d1
+    angular_hessian_.row(7).noalias() = Eigen::Vector4d((-sx * sy * cz), (sx * sy * sz), (sx * cy), 0.0); // d2
+    angular_hessian_.row(8).noalias() = Eigen::Vector4d((cx * sy * cz), (-cx * sy * sz), (-cx * cy), 0.0); // d3
 
-    angular_hessian_.row(9).noalias() = Eigen::Vector4d((sy * sz), (sy * cz), 0, 0.0f); // e1
-    angular_hessian_.row(10).noalias() = Eigen::Vector4d((-sx * cy * sz), (-sx * cy * cz), 0, 0.0f); // e2
-    angular_hessian_.row(11).noalias() = Eigen::Vector4d((cx * cy * sz), (cx * cy * cz), 0, 0.0f); // e3
+    angular_hessian_.row(9).noalias() = Eigen::Vector4d((sy * sz), (sy * cz), 0.0, 0.0f); // e1
+    angular_hessian_.row(10).noalias() = Eigen::Vector4d((-sx * cy * sz), (-sx * cy * cz), 0.0, 0.0); // e2
+    angular_hessian_.row(11).noalias() = Eigen::Vector4d((cx * cy * sz), (cx * cy * cz), 0.0, 0.0); // e3
 
-    angular_hessian_.row(12).noalias() = Eigen::Vector4d((-cy * cz), (cy * sz), 0, 0.0f); // f1
-    angular_hessian_.row(13).noalias() = Eigen::Vector4d((-cx * sz - sx * sy * cz), (-cx * cz + sx * sy * sz), 0, 0.0f); // f2
-    angular_hessian_.row(14).noalias() = Eigen::Vector4d((-sx * sz + cx * sy * cz), (-cx * sy * sz - sx * cz), 0, 0.0f); // f3
+    angular_hessian_.row(12).noalias() = Eigen::Vector4d((-cy * cz), (cy * sz), 0.0, 0.0); // f1
+    angular_hessian_.row(13).noalias() = Eigen::Vector4d((-cx * sz - sx * sy * cz), (-cx * cz + sx * sy * sz), 0.0, 0.0); // f2
+    angular_hessian_.row(14).noalias() = Eigen::Vector4d((-sx * sz + cx * sy * cz), (-cx * sy * sz - sx * cz), 0.0, 0.0); // f3
   }
 }
 
-
+// \note Thread-unsafe ```computePointDerivatives``` is deprecated
```

Comment:
Can we replace thread unsafe `computePointDerivatives` call with:
```cpp
void computePointDerivatives(const Eigen::Vector3d &x, bool compute_hessian)
{
  if (num_threads_ > 1) { PCL_WARN("This implementation is thread unsafe. Please use ..."); }
  computePointDerivatives(x, point_jacobian_, point_hessian_, compute_hessian);
}
```

This is better than a note which most ppl wouldn't read. It'd also reduce code duplication

Reply:
I replaced the thread unsafe implementations with calling thread safe ones